.github/workflows/smoke-test-package.yaml

```diff
@@ -0,0 +1,238 @@
+name: Smoke test package rpm/deb
+run-name: Smoke test package rpm/deb
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-version:
+        description: |
+          Override the build version from what is in the Makefile.
+          Use this override to specify a tag for build debugging purposes.
+        type: string
+
+  pull_request:
+  merge_group:
+
+jobs:
+  changes:
+    name: Check for relevant changes
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      changed: ${{ steps.changes.outputs.changed }}
+    steps:
+      - name: Checkout
+        if: ${{ github.event_name == 'merge_group' }}
+        uses: actions/checkout@v4
+      - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: changes
+        with:
+          base: ${{ github.event.pull_request.base.ref || github.event.merge_group.base_ref }}
+          ref: ${{ github.event.pull_request.head.ref || github.event.merge_group.head_ref }}
+          filters: |
+            changed:
+              - '.github/workflows/smoke-test-package.yaml'
+              - 'lib/autoupdate/**.go'
+              - 'tool/teleport-update/**.go'
+              - 'go.mod'
+              - 'go.sum'
+              - 'build.assets/Makefile'
+              - 'build.assets/Dockerfile*'
+              - 'Makefile'
+
+  build:
+    runs-on: ubuntu-latest
+    needs: changes
+    if: ${{ !startsWith(github.head_ref, 'dependabot/') && needs.changes.outputs.changed == 'true' }}
+    outputs:
+      version: ${{ steps.set-version.outputs.version }}
+    env:
+      # GOMODCACHE and GOCACHE can be removed when we commit to the self-hosted
+      # ARC runners, as these runners set up these vars for us. We still need
+      # these though for the GitHub runners (ubuntu).
```

Comment:
I'd remove this comment and the same one below on the step that uses these vars as it doesn't make sense in the context of the teleport repository - we do not have self-hosted runners available to the public repo.